tests/sentry/api/endpoints/test_sentry_app_details.py

```diff
@@ -80,3 +80,61 @@ def test_no_access_without_internal_catchall(self):
 
         response = self.client.get(self.url, format='json')
         assert response.status_code == 404
+
+
+class UpdateSentryAppDetailsTest(SentryAppDetailsTest):
+    @with_feature('organizations:internal-catchall')
+    def test_update_published_app(self):
+        self.login_as(user=self.superuser, superuser=True)
+        response = self.client.put(
+            self.url,
+            data={
+                'name': 'NewName',
+                'webhook_url': 'https://newurl.com',
+            },
+            format='json',
+        )
+        assert response.data == {
+            'name': 'NewName',
+            'scopes': [],
+            'uuid': self.published_app.uuid,
+            'webhook_url': 'https://newurl.com',
+        }
+
+    @with_feature('organizations:internal-catchall')
+    def test_update_unpublished_app(self):
+        self.login_as(user=self.user)
+        url = reverse('sentry-api-0-sentry-app-details', args=[self.unpublished_app.slug])
+
+        response = self.client.put(
+            url,
+            data={
+                'name': 'NewName',
+                'webhook_url': 'https://newurl.com',
+                'scopes': ('project:read',)
+            },
+            format='json',
+        )
+
+        assert response.status_code == 200
+        assert response.data == {
+            'name': 'NewName',
+            'scopes': ['project:read'],
+            'uuid': self.unpublished_app.uuid,
+            'webhook_url': 'https://newurl.com',
+        }
+
+    @with_feature('organizations:internal-catchall')
+    def test_cannot_update_scopes_published_app(self):
```

Comment:
@mnoble since we are going to be fine tuning the sentry app endpoint permissions should we make a note of this and do it then? Currently I don't think the endpoint will make this distinction

Reply:
It should prevent it currently. Right now it's using https://github.com/getsentry/sentry/blob/master/src/sentry/api/bases/sentryapps.py#L12 through the endpoint's baseclass.

Reply:
ahh right kk can add test for that